### PR TITLE
OpenShift template with notes. Updated README.md.

### PR DESCRIPTION
**Pull request by phracek**
Signed-off-by: Petr "Stone" Hracek <phracek@redhat.com>

### CODE REVIEW

**Comment by TomasTomecek** [comment]
I like this. Definitely useful. I'm personally interested in feedback from other team members on this (@eliskasl @asamalik @KarstenHopp): isn't the template too complicated? Are the image streams and triggers necessary? From my PoV, a simple specification of a service and/or pods would be enough. I might be completely wrong, so please explain the current solution.

File: openshift-template.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        type: Rolling
+      triggers:
+        - type: ImageChange
+          imageChangeParams:
+            automatic: true
+            containerNames:
+              - # Name of your container image. Taken from kind:ImageStream:name
+            from:
+              kind: ImageStreamTag
+              name: # Specify what version do you need. Usually the latest latest fedora:latest
```
Comment (TomasTomecek) [inline openshift-template.yml]: `latest latest`?

File: README.md [hunk 2]
```diff
@@ -14,6 +14,8 @@ There are two templates within this repository on two different branches:
 
  * `LICENSE` — pick the right license
  * `Makefile` — so it's easy to build, run, test
+ * `openshift-template.yml` - general OpenShift template
+  * update it. It includes several commands what to update
```
Comment (TomasTomecek) [inline README.md:18]: Could you please link to your real template from `postfix`, so people know how the template may look?

**Comment by phracek** [comment]
I would like to automate it a bit more. A script where user defines container name and may be pod name will automatically create working template for OpenShift. It can be automated like <container>-pod and <container_name>_deployment. This would be next step.

File: openshift-template.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            name: # I would prefer to use the same as template:metadata:name
+        spec:
+          containers: # Container specification
+            - env: # This is a list
+              - name: REDIS_PASSWORD
+                value: #define a value like ${REDIS_PASSWORD}
+              name: # Name. Use the same as ImageStream
+              image: # Use the same as kind:ImageStream:spec:dockerImageRepository
+              imagePullPolicy: Never # Never means, user will care about it. Deleting means, Update the image automaticaly
+              ports: # List which
```
Comment (TomasTomecek) [inline openshift-template.yml]: More elaborative comment would be useful: is this a port, within a container, on which the service is listening?

**Review by TomasTomecek — COMMENTED** [review]
This is definitely a great starting point. We can't cover all the use cases -- people need to understand how these should be written. Comments are just nits, nothing really important.

**Comment by TomasTomecek** [comment]
> I would like to automate it a bit more. A script where user defines container name and may be pod name will automatically create working template for OpenShift. It can be automated like -pod and <container_name>_deployment. This would be next step.

No idea how would you automate such thing when services listen on different ports, use different config files, store files in various locations and all behave differently. It could be possible to analyze image and maybe populate a template based on container image metadata (ports, volumes). Definitely worth spiking.

**Comment by phracek** [comment]
Yes, this can be taken from Dockerfile and I would like to do it. But I will wait for the other feedback.

**Review by TomasTomecek — APPROVED** [review]

**Pull request merged**